packages/web3-core-promievent/package.json

```diff
@@ -7,9 +7,12 @@
     "engines": {
         "node": ">=8.0.0"
     },
-    "main": "src/index.js",
+    "main": "dist/web3-core-promievent.cjs.js",
+    "module": "dist/web3-core-promievent.esm.js",
+    "unpkg": "dist/web3-core-promievent.min.js",
+    "jsdelivr": "dist/web3-core-promievent.min.js",
     "dependencies": {
-        "any-promise": "1.3.0",
-        "eventemitter3": "3.1.2"
+        "@babel/runtime": "^7.6.3",
```

Comment:
The ``@babel/runtime`` with his polyfills is required otherwise will the e2e browser test (which is using the latest chrome) failing. I've added it on some packages only because of the reason that any contributor and also me doesn't have to care about what JS features he is using or not. Duplicates will get removed during the bundling process of us (terser) or the developer who is using our library with a bundler and minification activated.

Reply:
> The @babel/runtime with his polyfills is required otherwise will the e2e browser test (which is using the latest chrome) failing.

Which pollyfills in particular are needed? IMO we should analyze each of them individually.

>  I've added it on some packages only because of the reason that any contributor and also me doesn't have to care about what JS features he is using or not.

I don't follow this. Why only some of the packages?

>  Duplicates will get removed during the bundling process of us (terser) or the developer who is using our library with a bundler and minification activated.

AFAIK `terser` doesn't remove duplicated code (except for duplicated switch cases). I see many people complaining of duplicated libraries in #1178, that's why I'm a little scared about this.

Reply:
> Which pollyfills in particular are needed? IMO we should analyze each of them individually. 

Babel does only add the required polyfills for the browsers who are listed from browserlist with the target ``last 2 versions``. There is no closer check to do from our standpoint. 

> I don't follow this. Why only some of the packages?

Some packages wouldn't require it but the ``@babel/runtime`` package is still existing and if a polyfill would be required from our code (for example if a contributor is using modern JS which isn't supported by all platforms) will Babel add it and otherwise not.

> AFAIK terser doesn't remove duplicated code (except for duplicated switch cases). I see many people complaining of duplicated libraries in #1178, that's why I'm a little scared about this.

Yes, it can remove duplications on the first "dependency layer" but doesn't detect them if they are lower in the dep tree. This is why I've added some additional rules to bring the bundle size down to 760kb. This can't get improved in this PR because it requires bigger changes to the package structure which would introduce some breaking changes. 
Cleaning up of the package structure can be done on ``1.3.0`` or even ``2.0``.

Reply:
[tree-shaking and dead code elimination explained](https://medium.com/@Rich_Harris/tree-shaking-versus-dead-code-elimination-d3765df85c80)